packages/botonic-plugin-flow-builder/src/action.tsx

```diff
@@ -18,31 +19,37 @@ export class FlowBuilderAction extends React.Component<FlowBuilderActionProps> {
     const flowBuilderPlugin = getFlowBuilderPlugin(request.plugins)
     const locale = flowBuilderPlugin.getLocale(request.session)
     const payload = request.input.payload
-    let targetContentId: string | undefined = payload
+    let targetNode: HtNodeWithContent | string | undefined = payload
+
     if (!payload && request.session.is_first_interaction) {
-      targetContentId = await flowBuilderPlugin.getStartId()
+      targetNode = flowBuilderPlugin.cmsApi.getStartNode()
     }
-    if (!payload) {
-      const intentPayload = await flowBuilderPlugin.getPayloadByIntent(
+
+    if (!payload && request.input.data) {
+      const intentNode = flowBuilderPlugin.cmsApi.getNodeByIntent(
         request.input,
         locale
       )
-      if (intentPayload) targetContentId = intentPayload
-      const keywordPayload = await flowBuilderPlugin.getPayloadByKeyword(
-        request.input,
+      const keywordNode = flowBuilderPlugin.cmsApi.getNodeByKeyword(
+        request.input.data,
         locale
       )
-      if (keywordPayload) targetContentId = keywordPayload
+      if (intentNode) {
+        targetNode = intentNode
+      } else if (keywordNode) {
+        targetNode = keywordNode
+      }
     }
-    if (!targetContentId) {
-      targetContentId = await flowBuilderPlugin.getFallbackId(
+
+    if (!targetNode) {
+      targetNode = flowBuilderPlugin.cmsApi.getFallbackNode(
         alternateFallbackMessage
```

Comment:
This value is always `false`. Is this parameter needed then?

Reply:
It will alternate between `true` and `false` but we have to fix the way this value is initialized because now it's shared with all users.